lib/kubeclient/common.rb

```diff
@@ -66,6 +71,15 @@ def initialize_client(
       end
     end
 
+    def method_missing(method_sym, *args, &block)
+      if !@discovered && method_sym.to_s.match(/^(get|watch|delete|create|update|patch)_(.*)$/)
```

Comment:
@moolitayer do we need to check for these or is it just an optimization that we could drop? Can we make a constant and simplify the check?

```ruby
ENTITY_METHODS = %w(get watch delete create update patch)

ENTITY_METHODS.any? { |x| method_sym.to_s.start_with?(x) }
```

I am not sure if I like the above better, anyway what I would like to have is a single explicit place where we can add these verbs.